app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java

```diff
@@ -330,7 +325,7 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
                     }
                 }
             }
-            if (onException.isPresent() && onExceptionStep != null) {
```

Comment:
Kind of a duplicated useless check. If onExceptionStep is not null, then onException is always present.

Reply:
You still have to check it as at line 321 `onException` is set to Optional.empty();